src/Enable.Extensions.Queuing.RabbitMQ/Internal/ConnectionManagement/RabbitMQConnectionManager.cs

```diff
@@ -0,0 +1,150 @@
+            _options = options;
+
+            _connectionFactory = new ConnectionFactory
+            {
+                HostName = options.HostName,
+                Port = options.Port,
+                VirtualHost = options.VirtualHost,
+                UserName = options.UserName,
+                Password = options.Password,
+                AutomaticRecoveryEnabled = options.AutomaticRecoveryEnabled,
+                NetworkRecoveryInterval = options.NetworkRecoveryInterval
+            };
+
+            _connectionHash = new
+            {
+                _options.HostName,
+                _options.Port,
+                _options.VirtualHost,
+                _options.UserName,
+                _options.Password,
+                _options.AutomaticRecoveryEnabled,
+                _options.NetworkRecoveryInterval,
+            }.GetHashCode();
+        }
+
+        /// <summary>
+        /// Provides a new RabbitMQ channel, session and model. The underlying connection is managed,
+        /// ensuring a connection is reused if possible.
+        /// </summary>
+        public RabbitMQManagedChannel CreateChannel()
+        {
+            RabbitMQManagedConnection managedConnection = null;
+            IModel channel = null;
+
+            managedConnection = GetConnection();
+
+            try
+            {
+                channel = managedConnection.Connection.CreateModel();
+            }
+            catch (Exception ex) when (ex is ChannelAllocationException || ex is AlreadyClosedException)
+            {
+                managedConnection.ChannelAllocationReached = true;
+
+                managedConnection = CreateConnection();
+                channel = managedConnection.Connection.CreateModel();
+            }
+
+            return new RabbitMQManagedChannel(channel, managedConnection.Id);
+        }
+
+        /// <summary>
+        /// Closes and disposes the channel. The underlying connection will be released and closed if no other clients are using it.
+        /// </summary>
+        public void CloseChannel(RabbitMQManagedChannel managedChannel)
+        {
+            var channel = managedChannel.Channel;
+
+            channel.Close();
+            channel.Dispose();
+
+            ReleaseConnection(managedChannel.ConnectionId);
+        }
+
+        /// <summary>
+        /// Provides access to the underlying RabbitMQ connection. Clients need not be concerned of the state of the connections available.
+        /// <see cref="ReleaseConnection"/> must be called when the connection is no longer required.
+        /// </summary>
+        private RabbitMQManagedConnection GetConnection()
+        {
+            lock (_lock)
+            {
+                var(hash, managedConnection) = _connections
+                    .Find(c => c.hash == _connectionHash
+                    && !c.managedConnection.ChannelAllocationReached);
+
+                if (managedConnection == null || managedConnection?.Connection == null)
```

Comment:
The check `managedConnection == null` is redundant, as the is covered by the following `managedConnection?.Connection == null` check.

Reply:
This has now been resolved.